src/DynamoCore/Nodes/CodeBlockNode.cs

```diff
@@ -310,14 +311,15 @@ internal override IEnumerable<AssociativeNode> BuildAst(List<AssociativeNode> in
 
             foreach (var astNode in codeStatements.Select(stmnt => NodeUtils.Clone(stmnt.AstNode)))
             {
-                MapIdentifiers(astNode);
+                if (context != AstBuilder.CompilationContext.NodeToCode)
+                    MapIdentifiers(astNode);
                 resultNodes.Add(astNode as AssociativeNode);
             }
 
             return resultNodes;
         }
 
-        public override IdentifierNode GetAstIdentifierForOutputIndex(int portIndex)
+        private IdentifierNode GetAstIdentifierForOutputIndexInternal(int portIndex, bool forRawName)
```

Comment:
I think `GetAstIdentifierForOutputIndexInternal` needs a little more `summary` and probably some explanation for the `forRawName` too (also possible return value of it). Can't quite seem to understand the purpose from reading the signature and debugging through its callers.